specification/under_development/Base-Profile.md

```diff
@@ -0,0 +1,160 @@
+# Base Profile
+
+This profile defines a subset of the QIR specification to support a coherent set of functionalities and capabilities that might be offered by a system. The Base Profile specifies the minimal requirements for executing quantum programs. Execution of a Base Profile compliant program does, for instance, not require support executing quantum instructions dependent on measurement outcomes. The only requirements for a backend are to be able to perform unitary transformations on the quantum state, as well as measurements at the end of the program.
+
+TODO: table with list of features that need to be supported by hardware, and what it doesn't need to support?  
+
+To support execution of Base Profile compliant programs, a QPU must have the following fundamental capabilities:
+- It supports executing a sequence of quantum instructions that permit to perform an arbitrary unitary transformation of the quantum state.
```

Comment:
This is a requirement on the _QPU_, rather than the QIR, and seems at odds with the idea of separating the instruction definitions from the profile, since if we go down that route there is no way to know from the QIR alone whether the instruction set is universal. It seems desirable to be able to determine whether some QIR is or is not base-profile-compliant without this being contingent on external definitions.